support/build/Makefile.rules

```diff
@@ -869,23 +869,23 @@ endef
 #  process. There are no global scopes for AWKINCLUDES and AWKFLAGS available
 #  because it would not be meaningful.
 define preprule_awk =
-$(3): $(2) \
-      $$($(call vprefix_lib,$(1),AWKINCLUDES)) $$($(call vprefix_lib,$(1),AWKINCLUDES-y)) \
-      $$($(call vprefix_src,$(1),$(2),,AWKINCLUDES)) $$($(call vprefix_src,$(1),$(2),,AWKINCLUDES-y)) \
-      | prepare
-	$(call build_cmd,AWK,$(1),$(3),\
-		$(AWK) -f $(2) \
-		       $$($(call vprefix_lib,$(1),AWKFLAGS)) $$($(call vprefix_lib,$(1),AWKFLAGS-y)) \
-		       $$($(call vprefix_src,$(1),$(2),,AWKFLAGS)) $$($(call vprefix_src,$(1),$(2),,AWKFLAGS-y)) \
-		       $(4) \
-		       $$($(call vprefix_lib,$(1),AWKINCLUDES)) $$($(call vprefix_lib,$(1),AWKINCLUDES-y)) \
-		       $$($(call vprefix_src,$(1),$(2),,AWKINCLUDES)) $$($(call vprefix_src,$(1),$(2),,AWKINCLUDES-y)) \
-		       > $(3)
-	)
-
-UK_SRCS-y += $(2)
-UK_PREPROCESS-y += $(3)
-$(eval $(call vprefix_lib,$(1),CLEAN-y) += $(call build_clean,$(3)))
+# $(3): $(2) \
```

Comment:
We used this rule to gather all the rust sources, but I'm now sure how it would work if you remove them.

Reply:
Yeah, this part only related with AWK file. And I enable all uklibs to compile unikraft, no error is reported. And I manually move this part into `lib/uklibid/Makefile.uk`.
The reason of removing them is mentioned in my commit message. Rust-bindgen file need uk/include/libid.h which generated from an awk file, but because `UK_PREPARE` is executed before `UK_PREPROCESS`, so one error that `not find  uk/include/libid.h` occurs.
In [issue#1147](https://github.com/unikraft/unikraft/issues/1147) is more detailed.

Reply:
I think this should be instead solved by stating the dependencies with the bingen rule. Please note that also external libraries are using these AWK rules. I assume `bindgen` should run just before a rust unit is compiled, so the rust compile rule should have bindgen as dependency (or actually the files it created) and bindgen should run after UK_PREPROCESS target was reached. So I think the bingen rule should have an [order-only pre-requisite](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html) to the preprocess target: `<...> | preprocess`
Would something like this work?